corehq/form_processor/backends/sql/processor.py

```diff
@@ -118,8 +124,11 @@ def save_processed_models(cls, processed_forms, cases=None, stock_result=None):
                 # Save deprecated form first to avoid ID conflicts
                 if processed_forms.deprecated:
                     FormAccessorSQL.update_form(processed_forms.deprecated, publish_changes=False)
+                    # in case of rollback, vault entry pk is cleared through tracked models on form
+                    save_tracked_vault_entries(processed_forms.deprecated)
 
                 FormAccessorSQL.save_new_form(processed_forms.submitted)
+                save_tracked_vault_entries(processed_forms.submitted)
```

Comment:
this seems to duplicate what's happening in `save_new_form`

Reply:
`save_new_form` is called in case of errors during processing or if its a `is_submission_error_log`, so we would need to do this at both places?
My understanding is that in an ideal form that gets processed smoothly, `save_processed_models` will take care of saving the vault entries and in case of any errors (or in submission error log, but we are not concerned about error logs) `save_new_form` will take care of saving vault entries. Is that not correct?

Reply:
but `save_processed_models`  already calls `save_new_form` (on the line above this)

Reply:
oh, missed that
https://github.com/dimagi/commcare-hq/pull/28161/commits/a3a216c27c829a8ac15b6b649c79f910013c1b26

I am now confused why we need to take care of transaction atomicity in this now then, specifically for vault entries since they are not saved here? Atomicity is taken care of new in `save_new_form` as well. 
Though I see, it is done here for cases/stock result/forms and other models as well. Guess that is for rollback in case of other errors before try block in `save_new_form`.